packages/terra-consumer-nav/src/components/nav-logo/NavLogo.jsx

```diff
@@ -41,10 +42,14 @@ const NavLogo = ({
   link,
   ...customProps
 }) => {
-  const image = (link) ?
-    <SmartLink {...link}> <img className={cx('img')} src={url} alt={altText} /> </SmartLink> :
-    <img className={cx('img')} src={url} alt={altText} />;
-  const body = (isCard && !!url) ? <Card.Body> {image} </Card.Body> : image;
+  const imgProps = {
+    className: cx('img'),
+    ...url && { src: url },
```

Comment:
I don't understand the `...url`. The spread operator is for objects, but url is a string

Reply:
the `...` is for the object. url is just the bool check where if it returns false, it appends nothing to the object. returning true, spreads the return object on the right side of the equation `{src: url}` onto the parent object (imgProps)

Reply:
it's also worth noting that it is able to be written this way since in the order of operations, `&&` has a higher precedence than `...`

Reply:
if url is falsey, what does it return?

Reply:
@zhongyn url is falsey you mean invalid url or empty ? 
if empty then `...url` will be false and src will not be added in imgProps. 
If invalid url then it will still be added but browser will render alt text in this case.

Reply:
@zhongyn if url is falsey, `...` essentially doesn't do anything, since it would be trying to spread a boolean 'false' which spreads nothing.

wrote a quick codepen if you want to play around with it: https://codepen.io/anon/pen/veEPpK?editors=1012

Reply:
there was also a question raised here: https://github.com/tc39/proposal-object-rest-spread/issues/45 in which it's stated that `...` mirrors `Object.assign` and
> Object.assign will treat all primitives (like undefined) as the same as {}

which therefore spreads nothing

Reply:
thats neat. seems tricky but i'm ok with it.